File: python/shared.py
# @@@SNIPSTART python-money-transfer-project-template-shared
from dataclasses import dataclass

PYTHON_DIMENSION_CALCULATION_TASK_QUEUE = "PYTHON_DIMENSION_CALCULATION_TASK_QUEUE"

@dataclass
class CalculationConfig:
    ProfileID: str
    ReferenceID: str


@dataclass
class ScoreProfile:
    SpaceID: str
    Name: str
    NodeType: str
    DimensionWeights: dict[str, float]


@dataclass
class Dimension:
    Name: str
    Value: float
    Explanation: str

File: golang/shared.go
package app

const ScoreCalculationTaskQueueName = "SCORE_CALCULATION_TASK_QUEUE"

type CalculationConfig struct {
	ProfileID   string
	ReferenceID string
}

type ScoreProfile struct {
	SpaceID          string
	Name             string
	NodeType         string
	DimensionWeights map[string]float64
}

type Dimension struct {
	Name        string
	Value       float64
	Explanation string
}

type Score struct {
	Score      float64
	Dimensions []Dimension
}
